File: backend/hotel-main/src/main/java/at/fhtw/hotel/model/Room.java
package at.fhtw.hotel.model;

import java.math.BigDecimal;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
@AllArgsConstructor
public class Room {
    private final Long id;
    private final String title;
    private final String description;
    private final int maxGuests;
    private final BigDecimal basePricePerNight;
    private final List<RoomImage> images;
    private final List<Extra> extras;
}

File: frontend/src/core/models/room.ts
export interface RoomImage {
  id: number;
  url: string;
  sortOrder: number;
  altText: string;
}

export interface Extra {
  id: number;
  code: string;
  title: string;
  description: string;
  iconName: string;
}

export interface Room {
  id: number;
  title: string;
  description: string;
  maxGuests: number;
  basePricePerNight: number;
  images: RoomImage[];
  extras: Extra[];
}
